test/e2e/snaps/test-snap-rpc.spec.js

```diff
@@ -0,0 +1,130 @@
+const { strict: assert } = require('assert');
+const { withFixtures } = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+const { TEST_SNAPS_WEBSITE_URL } = require('./enums');
+
+describe('Test Snap RPC', function () {
+  it('can use the cross-snap RPC endowment and produce a public key', async function () {
+    const ganacheOptions = {
+      accounts: [
+        {
+          secretKey:
+            '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+          balance: 25000000000000000000,
+        },
+      ],
+    };
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions,
+        failOnConsoleError: false,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+
+        // enter pw into extension
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // navigate to test snaps page and connect
+        await driver.driver.get(TEST_SNAPS_WEBSITE_URL);
+        await driver.delay(1000);
+        const snapButton1 = await driver.findElement('#connectRpcSnap');
+        await driver.scrollToElement(snapButton1);
+        await driver.delay(1000);
+        await driver.clickElement('#connectRpcSnap');
+        await driver.delay(1000);
+
+        // switch to metamask extension and click connect
+        let windowHandles = await driver.waitUntilXWindowHandles(
+          2,
+          1000,
+          10000,
+        );
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement(
+          {
+            text: 'Connect',
+            tag: 'button',
+          },
+          10000,
+        );
+
+        await driver.delay(2000);
+
+        // approve install of snap
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement({
+          text: 'Approve & install',
+          tag: 'button',
+        });
+
+        // delay for npm installation
+        await driver.delay(2000);
```

Comment:
Instead of adding an arbitrary delay, can we wait for the installation to complete? The installation may take longer sometimes and cause the test to fail. An example where this happened was when we were in the process of releasing v10.24.0. Most of the snap tests failed on the installation steps. Here is a [link](https://github.com/MetaMask/metamask-extension/pull/16840/commits/a464f128bc2ad95345e27812dba643d80d4335a6) to a change I made to ensure the install was complete without using a delay, I wait for the test dapp buttons to be updated.

Reply:
Fixed

Reply:
Fixed